tests/mir-opt/instcombine/ptr_cast.ptr_cast.InstCombine.diff

```diff
@@ -0,0 +1,23 @@
+- // MIR for `ptr_cast` before InstCombine
++ // MIR for `ptr_cast` after InstCombine
+  
+  fn ptr_cast(_1: *const u8) -> *mut () {
+      debug p => _1;                       // in scope 0 at $DIR/ptr_cast.rs:+0:17: +0:18
+      let mut _0: *mut ();                 // return place in scope 0 at $DIR/ptr_cast.rs:+0:34: +0:41
+      let mut _2: *mut u8;                 // in scope 0 at $DIR/ptr_cast.rs:+1:5: +1:17
+      let mut _3: *const u8;               // in scope 0 at $DIR/ptr_cast.rs:+1:5: +1:6
+  
+      bb0: {
+          StorageLive(_2);                 // scope 0 at $DIR/ptr_cast.rs:+1:5: +1:17
+          StorageLive(_3);                 // scope 0 at $DIR/ptr_cast.rs:+1:5: +1:6
+          _3 = _1;                         // scope 0 at $DIR/ptr_cast.rs:+1:5: +1:6
+-         _2 = move _3 as *mut u8 (PtrToPtr); // scope 0 at $DIR/ptr_cast.rs:+1:5: +1:17
++         _0 = move _3 as *mut () (PtrToPtr); // scope 0 at $DIR/ptr_cast.rs:+1:5: +1:28
++         nop;                             // scope 0 at $DIR/ptr_cast.rs:+1:16: +1:17
+          StorageDead(_3);                 // scope 0 at $DIR/ptr_cast.rs:+1:16: +1:17
+-         _0 = move _2 as *mut () (PtrToPtr); // scope 0 at $DIR/ptr_cast.rs:+1:5: +1:28
```

Comment:
Yay!  Getting casts like this simplified will help a whole bunch of stuff -- lots of the ptr metadata needs to cast multiple times.